Marlin/src/module/endstops.cpp

```diff
@@ -582,7 +591,7 @@ void Endstops::update() {
     #endif
   #endif
 
-  #if HAS_Z_MIN && !Z_SPI_SENSORLESS
+  #if HAS_Z_MIN && NONE(Z_SPI_SENSORLESS, Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN)
```

Comment:
AFAIK there is no reason for a Z probe result to end up in the Z2, Z3, and Z4 endstop states. But keep an eye out for any effect this change might have w/r/t multi-Z endstops / steppers.